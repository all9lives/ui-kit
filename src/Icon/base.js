import React, { Component } from 'react'
import Image from '../Image'
import Size from '../utils/Size'

const SIZE_MAPPING = {
  small: 0.5,
  regular: 1,
  large: 2
}

export class Base extends Component {
  render () {
    return (
      <Image
        {...this.props}
        size={SIZE_MAPPING[this.props.size] * Size.icon}
      />
    )
  }
}
