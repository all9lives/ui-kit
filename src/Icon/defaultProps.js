import Size from '../utils/Size'

export default {
  size: Size.regular
}
