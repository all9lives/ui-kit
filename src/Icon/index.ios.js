import Icon from './index.android'

export default Icon
