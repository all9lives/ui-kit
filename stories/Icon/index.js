import React from 'react'
import createStory from '../../storybook/createStory'
import Icon from '../../src/Icon'

const sourceUrl = 'https://avatars0.githubusercontent.com/u/6601073?v=4&s=40'

createStory('Icon', module)
  .add('default', () => <Icon source={sourceUrl} />)
  .add('with size `large`', () => <Icon size='large' source={sourceUrl} />)
