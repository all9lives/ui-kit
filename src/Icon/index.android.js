import styled from 'styled-components/native'
import base, { attrs } from './base'
import defaultProps from './defaultProps'
import Image from '../Image'

const Icon = styled(Image).attrs(attrs)`
  ${base}
`
Icon.defaultProps = defaultProps

export default Icon
