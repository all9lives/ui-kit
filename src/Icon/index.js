import { Base as Icon } from './base'
import defaultProps from './defaultProps'

Icon.defaultProps = defaultProps

export default Icon
