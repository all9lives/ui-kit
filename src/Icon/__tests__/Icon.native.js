/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import Icon from '../index.ios.js'

describe('Icon.native', () => {
  it('renders a snapshot', () => {
    const wrapper = shallow(
      <Icon src='any source' />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `size:large`', () => {
    const wrapper = shallow(
      <Icon src='any source' size='large' />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
