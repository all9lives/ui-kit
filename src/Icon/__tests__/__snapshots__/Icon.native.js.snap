// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon.native renders a snapshot 1`] = `
<mapProps(Styled(Image))
  size={24}
  src="any source"
/>
`;

exports[`Icon.native renders a snapshot with \`size:large\` 1`] = `
<mapProps(Styled(Image))
  size={48}
  src="any source"
/>
`;

exports[`Icon.native renders a snapshot with \`size:small\` 1`] = `
<mapProps(Styled(Image))
  size={12}
  src="any source"
/>
`;
